src/interchange/src/protobuf.rs

```diff
@@ -92,25 +102,68 @@ impl DecodedDescriptors {
 /// Decodes a particular Protobuf message from its wire format.
 #[derive(Debug)]
 pub struct Decoder {
-    message_descriptor: MessageDescriptor,
+    descriptors: DecodedDescriptors,
     packer: Row,
+    ccsr_client: Option<ccsr::Client>,
+    validated_schema_id: Option<i32>,
 }
 
 impl Decoder {
     /// Constructs a decoder for a particular Protobuf message.
-    pub fn new(descriptors: DecodedDescriptors) -> Self {
+    pub fn new(
+        descriptors: DecodedDescriptors,
+        schema_registry: Option<ccsr::ClientConfig>,
+    ) -> Self {
         Decoder {
-            message_descriptor: descriptors.message_descriptor,
-            packer: Row::default(),
+            descriptors,
+            packer: Default::default(),
+            ccsr_client: schema_registry.map(|sr| sr.build()).transpose().unwrap(),
```

Comment:
`unwrap` here doesn't look safe! I guess we do it in the Avro decoder but that's terrible. Can you at least push the call to `expect` here up through the stack into `dataflow::src::decode::get_decoder` so that at least it's in the same place as it is for the Avro path?

Reply:
oh this is definitely an error case I am i missing!

Reply:
with the new commit, resolving this at a higher level is here: https://github.com/MaterializeInc/materialize/issues/9613